pkg/commands/workload.go

```diff
@@ -323,9 +332,13 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
 	if !okToPush {
 		return okToPush, nil
 	}
+	localPathDir := opts.LocalPath
+	if opts.SubPath != "" {
+		localPathDir = filepath.Join(localPathDir, opts.SubPath)
+	}
```

Comment:
this will end up applying the subpath twice, once when uploading the source, and again when building it the supply chain.

Reply:
This change is probably unnecessary then—the supply chain to build the image based on the sub-path of the source directory.